### PR TITLE
LoopLoadElim: re-org tests after invalid #96656

### PR DESCRIPTION
**Pull request by artagnon**
After pr96656.ll were added to LAA and LoopVersioning, it was decided that the bug is in a caller of LoopVersioning, not in LAA or LoopVersioning itself. The new candidate was LoopLoadElim, but #96656 has since been marked invalid. Hence, re-organize the added tests to avoid confusion, and the testcase from the investigation to LoopLoadElim.

### CODE REVIEW

**Comment by artagnon** [comment]
Gentle ping.

**Comment by artagnon** [comment]
Gentle ping.

**Comment by artagnon** [comment]
This patch is now a clean-up and re-org, since #96656 is now invalid.

**Comment by artagnon** [comment]
Gentle ping. This is a straightforward cleanup.

**Comment by artagnon** [comment]
Gentle ping.

File: llvm/test/Transforms/LoopVersioning/single-iteration.ll [hunk 1]
```diff
@@ -1,16 +1,19 @@
 ; NOTE: Assertions have been autogenerated by utils/update_test_checks.py UTC_ARGS: --version 5
 ; RUN: opt -passes=loop-versioning -S %s | FileCheck %s
 
-define void @lver.check.unnecessary(ptr %arg, ptr %arg1, i1 %arg2) {
-; CHECK-LABEL: define void @lver.check.unnecessary(
-; CHECK-SAME: ptr [[ARG:%.*]], ptr [[ARG1:%.*]], i1 [[ARG2:%.*]]) {
+; Callers should not call LoopVersioning on single-iteration loops, but
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVersioning/single-iteration.ll:4]: Would be good to explain why, e.g.
```suggestion
; Callers should not call LoopVersioning on single-iteration loops as it is very likely not profitable
```

**Review by fhahn — APPROVED** [review]
LGTM, thanks!

**Pull request merged**